test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodLongHaulReportGeneratorTest.cs

```diff
@@ -0,0 +1,394 @@
+        {
+            int batchSize = 10;
+            var mockSenderResults = new Mock<ITestResultCollection<TestOperationResult>>();
+            var mockReceiverStore = new Mock<ISequentialStore<TestOperationResult>>();
+            var receiverResults = Option.Some<ITestResultCollection<TestOperationResult>>(
+               new StoreTestResultCollection<TestOperationResult>(mockReceiverStore.Object, batchSize));
+
+            ArgumentException ex = Assert.Throws<ArgumentException>(
+                () => new DirectMethodLongHaulReportGenerator(
+                    TestDescription,
+                    trackingId,
+                    "senderSource",
+                    mockSenderResults.Object,
+                    Option.Some("receiverSource"),
+                    receiverResults,
+                    "resultType1"));
+
+            Assert.StartsWith("trackingId", ex.Message);
+        }
+
+        [Theory]
+        [MemberData(nameof(DirectMethodLongHaulReportData.GetCreateReportData), MemberType = typeof(DirectMethodLongHaulReportData))]
+        public async Task TestCreateReportAsync(
+            IEnumerable<ulong> senderStoreValues,
+            IEnumerable<ulong> receiverStoreValues,
+            IEnumerable<HttpStatusCode> statusCodes,
+            IEnumerable<DateTime> timestamps,
+            int batchSize,
+            bool expectedIsPassed,
+            ulong expectedOk,
+            ulong expectedStatusCodeZero,
+            ulong expectedUnknown)
+        {
+            string senderSource = "senderSource";
+            string receiverSource = "receiverSource";
+            string resultType = TestOperationResultType.DirectMethod.ToString();
+
+            var mockSenderStore = new Mock<ISequentialStore<TestOperationResult>>();
+            var senderResults = new StoreTestResultCollection<TestOperationResult>(mockSenderStore.Object, batchSize);
+            var mockReceiverStore = new Mock<ISequentialStore<TestOperationResult>>();
+            var receiverResults = Option.Some<ITestResultCollection<TestOperationResult>>(
+                new StoreTestResultCollection<TestOperationResult>(mockReceiverStore.Object, batchSize));
+
+            var reportGenerator = new DirectMethodLongHaulReportGenerator(
+                TestDescription,
+                Guid.NewGuid().ToString(),
+                senderSource,
+                senderResults,
+                Option.Some(receiverSource),
+                receiverResults,
+                resultType);
+
+            Guid guid = Guid.NewGuid();
+
+            var senderStoreData = GetSenderStoreData(senderSource, resultType, senderStoreValues, statusCodes, timestamps, guid);
+            for (int i = 0; i < senderStoreData.Count; i += batchSize)
+            {
+                int startingOffset = i;
+                mockSenderStore.Setup(s => s.GetBatch(startingOffset, batchSize)).ReturnsAsync(senderStoreData.Skip(startingOffset).Take(batchSize));
```

Comment:
Do we really use all these setups or is this just for safety?

Reply:
We actually use them when creating reports

Reply:
What I meant was that it seems like it should take elements in batch sizes starting from each index, which I would guess leads to duplicates. So I was assuming it doesn't use all these setup calls. Regardless I think it is fine.